graph/appender/dead_node.go

```diff
@@ -123,28 +122,33 @@ func applyDeadNodes(trafficMap graph.TrafficMap, globalInfo *GlobalInfo, namespa
 	}
 }
 
-// resolveExternalServices queries the cluster API to resolve egress services
+// isExternalService queries the cluster API to resolve egress services
 // by using ServiceEntries resources across all namespaces in the cluster.
-// All ServiceEntries are needed, because Istio does not distinguish where the
-// ServiceEntries are created when routing egress traffic.
-func isExternalService(service string, namespaceInfo *NamespaceInfo, globalInfo *GlobalInfo) bool {
+// All ServiceEntries are needed because Istio does not distinguish where the
+// ServiceEntries are created when routing egress traffic (i.e. a ServiceEntry
+// can be in any namespace and it will still work).
+// However, an egress service will have its namespace set to "default" in the telemetry.
+func (a DeadNodeAppender) isExternalService(service string, namespaceInfo *NamespaceInfo, globalInfo *GlobalInfo) bool {
 	if namespaceInfo.ExternalServices == nil {
 		namespaceInfo.ExternalServices = make(map[string]bool)
 
-		// Currently no other appenders use ServiceEntries, so they are not cached in NamespaceInfo
-		istioCfg, err := globalInfo.Business.IstioConfig.GetIstioConfigList(business.IstioConfigCriteria{
-			IncludeServiceEntries: true,
-			Namespace:             namespaceInfo.Namespace,
-		})
-		checkError(err)
-
-		for _, entry := range istioCfg.ServiceEntries {
-			if entry.Spec.Hosts != nil && entry.Spec.Location == "MESH_EXTERNAL" {
-				for _, host := range entry.Spec.Hosts.([]interface{}) {
-					namespaceInfo.ExternalServices[host.(string)] = true
+		for ns := range a.AccessibleNamespaces {
```

Comment:
I'm wondering how to make this work in the _multi-tenancy_ or _reduced privileges_ use cases.

Since external services * just work *, we should look for them in all namespaces (not only accessible ones). But Kiali is running with reduced permissions... So... well... I don't know. This is just a comment to have this point in mind.

Reply:
We can only look at what is accessible to the user. So he will only see accessible namespaces, but if restricted access is given to the user, then this will only show what the user can see, which is basically "working as expected" in that case.

Reply:
I know. That's OK from that point of view. However, workloads will still be able to use the *external services*.

I mean... External services needs to be defined somewhere. But they really don't belong to a namespace... they are... well... external to the mesh. If we hide them just because the definition is in namespace that the user can't access doesn't make much sense to me.

Reply:
I agree with @israel-hdez , it doesn't make a lot of sense that we can show the node but not be able to flag it as an egress node.  I feel like we're going to have to revisit this going forward.  As it stands now we can only flag egress nodes defined in a namespace kiali can access.